core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Collector.scala

```diff
@@ -123,12 +124,15 @@ trait Collector {
 
     lazy val redirectRoutes =
       scheme("http") {
-        extract(_.request.uri) { uri =>
-          redirect(
-            uri.copy(scheme = "https").withPort(collectorConf.ssl.port),
-            StatusCodes.MovedPermanently
-          )
-        }
+        redirectTo(Scheme.Https, collectorConf.ssl.port, StatusCodes.MovedPermanently)
+      }
+
+    def redirectTo(scheme: Scheme, port: Int, redirectionType: StatusCodes.Redirection) =
```

Comment:
I'm curious, why do you choose to pass `redirectionType` as a parameter if it can only ever take one value?

Reply:
I guess it is a premature optimization, thinking for future use where we might need different status code